src/main/java/org/folio/rest/support/EffectiveCallNumberComponentsUtil.java

```diff
@@ -6,12 +6,25 @@
 import org.folio.rest.jaxrs.model.EffectiveCallNumberComponents;
 import org.folio.rest.jaxrs.model.HoldingsRecord;
 import org.folio.rest.jaxrs.model.Item;
+import org.folio.services.CallNumberUtils;
+
+import java.util.Optional;
 
 public final class EffectiveCallNumberComponentsUtil {
   private EffectiveCallNumberComponentsUtil() {}
 
   public static void setCallNumberComponents(Item item, HoldingsRecord hr) {
     item.setEffectiveCallNumberComponents(buildComponents(item, hr));
+    setShelvesNumber(item);
```

Comment:
I think whether this belongs here depends upon whether the `shelving order` is of the effective call number components or separate.

If it is separate, I think this should be moved to it's own class.

Reply:
I believe the domain folks have now [answered](https://issues.folio.org/browse/MODINVSTOR-381?focusedCommentId=97850&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-97850) that question, the working out of shelving order should not be part of working out the call number components.